test/v200/utils/common/command_test_utils.go

```diff
@@ -0,0 +1,212 @@
+package common
+
+import (
+	"fmt"
+
+	schema "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+)
+
+// commandAdded adds a new command to the test schema data and notifies the follower
+func (testDevFile *TestDevfile) commandAdded(command schema.Command) {
+	LogInfoMessage(fmt.Sprintf("command added Id: %s", command.Id))
+	testDevFile.SchemaDevFile.Commands = append(testDevFile.SchemaDevFile.Commands, command)
+	if testDevFile.Follower != nil {
+		testDevFile.Follower.AddCommand(command)
+	}
+}
+
+// commandUpdated and notifies the follower of the command which has been updated
+func (testDevFile *TestDevfile) commandUpdated(command schema.Command) {
+	LogInfoMessage(fmt.Sprintf("command updated Id: %s", command.Id))
+	if testDevFile.Follower != nil {
+		testDevFile.Follower.UpdateCommand(command)
+	}
+}
+
+// addEnv creates and returns a specifed number of env attributes in a schema structure
+func addEnv(numEnv int) []schema.EnvVar {
+	commandEnvs := make([]schema.EnvVar, numEnv)
+	for i := 0; i < numEnv; i++ {
+		commandEnvs[i].Name = "Name_" + GetRandomString(5, false)
+		commandEnvs[i].Value = "Value_" + GetRandomString(5, false)
+		LogInfoMessage(fmt.Sprintf("Add Env: %s", commandEnvs[i]))
+	}
+	return commandEnvs
+}
+
+// addAttributes creates returns a specifed number of attributes in a schema structure
+func addAttributes(numAtrributes int) map[string]string {
+	attributes := make(map[string]string)
+	for i := 0; i < numAtrributes; i++ {
+		AttributeName := "Name_" + GetRandomString(6, false)
+		attributes[AttributeName] = "Value_" + GetRandomString(6, false)
+		LogInfoMessage(fmt.Sprintf("Add attribute : %s = %s", AttributeName, attributes[AttributeName]))
+	}
+	return attributes
+}
+
+// addGroup creates and returns a group in a schema structure
+func (testDevFile *TestDevfile) addGroup() *schema.CommandGroup {
+
+	commandGroup := schema.CommandGroup{}
+	commandGroup.Kind = GetRandomGroupKind()
+	LogInfoMessage(fmt.Sprintf("group Kind: %s, default already set %t", commandGroup.Kind, testDevFile.GroupDefaults[commandGroup.Kind]))
+	// Ensure only one and at least one of each type are labelled as default
+	if !testDevFile.GroupDefaults[commandGroup.Kind] {
+		testDevFile.GroupDefaults[commandGroup.Kind] = true
+		commandGroup.IsDefault = true
+	} else {
+		commandGroup.IsDefault = false
+	}
+	LogInfoMessage(fmt.Sprintf("group isDefault: %t", commandGroup.IsDefault))
+	return &commandGroup
+}
+
+// AddCommand creates a command of a specified type in a schema structure and pupulates it with random attributes
+func (testDevFile *TestDevfile) AddCommand(commandType schema.CommandType) schema.Command {
+
+	var command *schema.Command
+	if commandType == schema.ExecCommandType {
```

Comment:
consider to use a switch case instead. gonna be more clear

Reply:
Not a big fan of switch statements unless there are a lot of else statements, but will consider changing if it is more clear.

Reply:
Fixed in new commit